pipeline/stage02_processing/scripts/spatial_downsampling.py

```diff
@@ -31,12 +25,17 @@ def spatial_smoothing(images, macro_pixel_dim):
                                    file_origin=images.file_origin,
                                    **imgseq.annotations)
 
-    imgseq_reduced.name += " "
+    imgseq_reduced.name = images.name + " "
     imgseq_reduced.annotations.update(macro_pixel_dim=macro_pixel_dim)
-    imgseq_reduced.description += "spatially downsampled ({})." .format(os.path.basename(__file__))
+    imgseq_reduced.description = images.name +  "spatially downsampled ({})." .format(os.path.basename(__file__))
 
     return imgseq_reduced
 
+def plot_downsampled_image(images_reduced):
+    plt.figure()
+    plt.imshow(np.array(images_reduced[0]), interpolation='nearest', cmap='viridis', origin='lower')
+    save_plot(args.output_img)
+
```

Comment:
Sorry if my previous comment was unclear or misleading. What I mean by that any function should use ideally not more or fewer arguments than provided in the definition, would look something like that:
```
def plot_downsampled_image(image, output_path):
    plt.figure()
    plt.imshow(image), interpolation='nearest', cmap='viridis', origin='lower')
    save_plot(output_path)
```
which can be then called as `plot_downsampled_image(images_reduced.as_array()[0], args.output_img)`

Reply:
Yeah, sorry I didn't get it!